### PR TITLE
Clouds

### PR DESCRIPTION
**Pull request by VZout**
Modified version of IQ's clouds. 
https://www.shadertoy.com/view/XslGRr

tinyclouds

### CODE REVIEW

File: examples/runners/wgpu/src/graphics.rs [hunk 1]
```diff
@@ -115,7 +115,7 @@ async fn run(
         // the resources are properly cleaned up.
         let _ = (&instance, &adapter, &module, &pipeline_layout);
 
-        *control_flow = ControlFlow::Wait;
+        *control_flow = ControlFlow::Poll;
```
Comment (Jasper-Bekkers) [inline examples/runners/wgpu/src/graphics.rs:118]: This got changed to Wait on the android PR, why are we changing it back?

Comment (VZout) [reply examples/runners/wgpu/src/graphics.rs:118]: If you put it to wait the window only gets updated when you move the mouse or other events get triggered.

**Comment by XAMPPRocky** [comment]
Trying to run this on the WGPU runner on macOS 11.0.1 gives the following error.

```
[0.187509 ERROR]()(no module): Error compiling the shader CompilationFailed("Compilation failed: \n\nprogram_source:689:36: error: call to \'mix\' is ambiguous\n    return (-1.0) + (2.0 * mix(mix(mix(hash(_1615), hash(_1616), _1698.x), mix(hash(_1617), hash(_1618), _1698.x), _1698.y), mix(mix(hash(_1619), hash(_1620), _1698.x), mix(hash(_1621), hash(_1622), _1698.x), _1698.y), _1698.z));\n                                   ^~~\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:406:18: note: candidate function\nMETAL_FUNC float mix(float x, float y, float a)\n                 ^\nprogram_source:529:7: note: candidate function\nfloat mix(float _1606, float _1607, float _1608)\n      ^\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:300:17: note: candidate function\nMETAL_FUNC half mix(half x, half y, half a)\n                ^\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:317:18: note: candidate function\nMETAL_FUNC half2 mix(half2 x, half2 y, half2 a)\n                 ^\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:334:18: note: candidate function\nMETAL_FUNC half3 mix(half3 x, half3 y, half3 a)\n                 ^\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:351:18: note: candidate function\nMETAL_FUNC half4 mix(half4 x, half4 y, half4 a)\n                 ^\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:423:19: note: candidate function\nMETAL_FUNC float2 mix(float2 x, float2 y, float2 a)\n                  ^\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:440:19: note: candidate function\nMETAL_FUNC float3 mix(float3 x, float3 y, float3 a)\n                  ^\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:457:19: note: candidate function\nMETAL_FUNC float4 mix(float4 x, float4 y, float4 a)\n                  ^\nprogram_source:689:76: error: call to \'mix\' is ambiguous\n    return (-1.0) + (2.0 * mix(mix(mix(hash(_1615), hash(_1616), _1698.x), mix(hash(_1617), hash(_1618), _1698.x), _1698.y), mix(mix(hash(_1619), hash(_1620), _1698.x), mix(hash(_1621), hash(_1622), _1698.x), _1698.y), _1698.z));\n                                                                           ^~~\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:406:18: note: candidate function\nMETAL_FUNC float mix(float x, float y, float a)\n                 ^\nprogram_source:529:7: note: candidate function\nfloat mix(float _1606, float _1607, float _1608)\n      ^\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:300:17: note: candidate function\nMETAL_FUNC half mix(half x, half y, half a)\n                ^\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:317:18: note: candidate function\nMETAL_FUNC half2 mix(half2 x, half2 y, half2 a)\n                 ^\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:334:18: note: candidate function\nMETAL_FUNC half3 mix(half3 x, half3 y, half3 a)\n                 ^\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:351:18: note: candidate function\nMETAL_FUNC half4 mix(half4 x, half4 y, half4 a)\n                 ^\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:423:19: note: candidate function\nMETAL_FUNC float2 mix(float2 x, float2 y, float2 a)\n                  ^\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:440:19: note: candidate function\nMETAL_FUNC float3 mix(float3 x, float3 y, float3 a)\n                  ^\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:457:19: note: candidate function\nMETAL_FUNC float4 mix(float4 x, float4 y, float4 a)\n                  ^\nprogram_source:689:130: error: call to \'mix\' is ambiguous\n    return (-1.0) + (2.0 * mix(mix(mix(hash(_1615), hash(_1616), _1698.x), mix(hash(_1617), hash(_1618), _1698.x), _1698.y), mix(mix(hash(_1619), hash(_1620), _1698.x), mix(hash(_1621), hash(_1622), _1698.x), _1698.y), _1698.z));\n                                                                                                                                 ^~~\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:406:18: note: candidate function\nMETAL_FUNC float mix(float x, float y, float a)\n                 ^\nprogram_source:529:7: note: candidate function\nfloat mix(float _1606, float _1607, float _1608)\n      ^\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:300:17: note: candidate function\nMETAL_FUNC half mix(half x, half y, half a)\n                ^\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:317:18: note: candidate function\nMETAL_FUNC half2 mix(half2 x, half2 y, half2 a)\n                 ^\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:334:18: note: candidate function\nMETAL_FUNC half3 mix(half3 x, half3 y, half3 a)\n                 ^\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:351:18: note: candidate function\nMETAL_FUNC half4 mix(half4 x, half4 y, half4 a)\n                 ^\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:423:19: note: candidate function\nMETAL_FUNC float2 mix(float2 x, float2 y, float2 a)\n                  ^\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:440:19: note: candidate function\nMETAL_FUNC float3 mix(float3 x, float3 y, float3 a)\n                  ^\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:457:19: note: candidate function\nMETAL_FUNC float4 mix(float4 x, float4 y, float4 a)\n                  ^\nprogram_source:689:170: error: call to \'mix\' is ambiguous\n    return (-1.0) + (2.0 * mix(mix(mix(hash(_1615), hash(_1616), _1698.x), mix(hash(_1617), hash(_1618), _1698.x), _1698.y), mix(mix(hash(_1619), hash(_1620), _1698.x), mix(hash(_1621), hash(_1622), _1698.x), _1698.y), _1698.z));\n                                                                                                                                                                         ^~~\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:406:18: note: candidate function\nMETAL_FUNC float mix(float x, float y, float a)\n                 ^\nprogram_source:529:7: note: candidate function\nfloat mix(float _1606, float _1607, float _1608)\n      ^\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:300:17: note: candidate function\nMETAL_FUNC half mix(half x, half y, half a)\n                ^\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:317:18: note: candidate function\nMETAL_FUNC half2 mix(half2 x, half2 y, half2 a)\n                 ^\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:334:18: note: candidate function\nMETAL_FUNC half3 mix(half3 x, half3 y, half3 a)\n                 ^\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:351:18: note: candidate function\nMETAL_FUNC half4 mix(half4 x, half4 y, half4 a)\n                 ^\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:423:19: note: candidate function\nMETAL_FUNC float2 mix(float2 x, float2 y, float2 a)\n                  ^\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:440:19: note: candidate function\nMETAL_FUNC float3 mix(float3 x, float3 y, float3 a)\n                  ^\n/System/Library/PrivateFrameworks/GPUCompiler.framework/Versions/31001/Libraries/lib/clang/31001.152/include/metal/metal_common:457:19: note: candidate function\nMETAL_FUNC float4 mix(float4 x, float4 y, float4 a)\n                  ^\n")
thread 'main' panicked at 'failed to create graphics pipeline: Failed to create pipeline: Unsupported usage: Implementation specific error occurred', /Users/g.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-core-0.6.5/src/device/mod.rs:2898:30
```

**Comment by VZout** [comment]
Fixed the apple thing as long as they don't define lerp as well.

**Comment by khyperia** [comment]
Linking to https://github.com/EmbarkStudios/rust-gpu/issues/102 because we hit that again.

**Comment by XAMPPRocky** [comment]
Going to close this for now, as this entire PR needs to be rebased, and I'm not sure we actually want to merge it (?) Feel free to re-open/open a new PR if you think should be merged though!

**Pull request closed without merging**